test/OpenTelemetry.Tests/Implementation/Trace/Propagation/CompositePropagatorTest.cs

```diff
@@ -0,0 +1,146 @@
+﻿// <copyright file="CompositePropagatorTest.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using OpenTelemetry.Context.Propagation;
+using Xunit;
+
+namespace OpenTelemetry.Tests.Implementation.Trace.Propagation
+{
+    public class CompositePropagatorTest
+    {
+        private const string TraceParent = "traceparent";
+        private static readonly string[] Empty = new string[0];
+        private static readonly Func<IDictionary<string, string>, string, IEnumerable<string>> Getter = (headers, name) =>
+        {
+            if (headers.TryGetValue(name, out var value))
+            {
+                return new[] { value };
+            }
+
+            return Empty;
+        };
+
+        private static readonly Action<IDictionary<string, string>, string, string> Setter = (carrier, name, value) =>
+        {
+            carrier[name] = value;
+        };
+
+        private ActivityTraceId traceId = ActivityTraceId.CreateRandom();
+        private ActivitySpanId spanId = ActivitySpanId.CreateRandom();
+
+        [Fact]
+        public void CompositePropagator_EmptyTextFormatList()
+        {
+            Dictionary<string, string> expected = new Dictionary<string, string>();
+            Dictionary<string, string> carrier = new Dictionary<string, string>();
+            var compositePropagator = new CompositePropagator();
+            var activityContext = compositePropagator.Extract(default, carrier, Getter);
+            Assert.Equal(default, activityContext);
+
+            activityContext = new ActivityContext(this.traceId, this.spanId, ActivityTraceFlags.Recorded, traceState: null);
+            compositePropagator.Inject(activityContext, carrier, Setter);
+            Assert.Equal(expected, carrier);
+
+            bool isInjected = compositePropagator.IsInjected(carrier, Getter);
+            Assert.True(isInjected);
```

Comment:
It feels kind of strange that it returns `true` for `IsInjected` when empty. If `IsInjected` resolves `true` you would expect `Extract` would return a valid context right? Not sure what to do about it. Kind of second-guessing the `IsInjected` API. I would say let's remove it if we didn't need it for the http instrumentation. We could have `ShouldInject` and `CanExtract` but that is heavy for what is essentially an edge case?

Reply:
since is just for this edge case, i would leave this for now, so we get compliant with the spec and think about what would be the best for us, since that isn't specified in the spec and, we could ask the community what they think.

Reply:
Yeah, I agree it's odd that we return true when the text format list is empty. Maybe we could/should return false if the list of formats is empty?

Reply:
i'm updating :)